internal/handlers/tigris/tigrisdb/databases.go

```diff
@@ -37,19 +39,32 @@ func (tdb *TigrisDB) createDatabaseIfNotExists(ctx context.Context, db string) (
 
 	// Database does not exist. Try to create it,
 	// but keep in mind that it can be created in concurrent connection.
-	err = tdb.Driver.CreateDatabase(ctx, db)
-	switch err := err.(type) {
-	case nil:
-		return true, nil
-	case *driver.Error:
-		if IsAlreadyExists(err) {
-			return false, nil
-		}
+	// If we detect that other creation is in flight, we give up to three attempts to create the database.
```

Comment:
Do we want to keep it, or we want Tigris to fix it?

Reply:
I'm afraid we have to keep it. Tigris said they would fix the error code (so it's not "unknown" anymore). But as far as I understand, they consider such a situation possible - the DB is in the creation process from some other request, and we don't know yet if it'd be created successfully. So we need another attempt to understand how it went.

Reply:
My understanding is that their client would do retrying transparently for us.

Let's create an issue for that in our repo and add TODO there

Reply:
added